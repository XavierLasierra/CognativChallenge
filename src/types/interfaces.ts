export interface IRecipe {
  _id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  duration: number;
  complexity: string;
  favorite?: number;
  people: number;
  recommended?: number;
  ingredients: string;
  description: string;
  photo?: string;
}

export interface IRecipePreviewProps {
  recipe: IRecipe;
  actionOnPress: (recipe: string) => void;
}

export interface IRecipeSummaryProps {
  recipe: IRecipe;
  alignCenter?: boolean;
}

interface IParams {
  recipeId: string;
}
interface IRoute {
  params: IParams;
}
export interface IRecipeDetailsProps {
  route: IRoute;
  navigation: INavigation;
}

interface INavigation {
  push: (path: string, params: IParams) => void;
  pop: () => void;
}

export interface IHomeProps {
  navigation: INavigation;
}

export interface IRecipeProp {
  item: IRecipe;
}

export interface IBackButtonProps {
  actionOnPress: () => void;
}

interface IRecipeData {
  data: IRecipe[];
}

export interface IAxiosRecipeData {
  data: IRecipeData;
}
