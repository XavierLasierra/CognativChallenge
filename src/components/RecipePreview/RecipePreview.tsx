import React from "react";
import { Image, TouchableOpacity, View } from "react-native";
import { IRecipePreviewProps } from "../../types/interfaces";

import placeholderConstants from "../../constants/placeholder.constants";

import styles from "./recipePreview.styles";
import RecipeSummary from "../RecipeSummary/RecipeSummary";

export default function RecipePreview({
  recipe,
  actionOnPress,
}: IRecipePreviewProps) {
  return (
    <TouchableOpacity onPress={() => actionOnPress(recipe)}>
      <View style={styles.rowContainer}>
        <View style={styles.imageContainer}>
          <Image
            source={{
              uri: recipe?.photo || placeholderConstants.defaultImageUrl,
            }}
            style={styles.recipeImage}
          />
        </View>
        <RecipeSummary recipe={recipe} />
      </View>
    </TouchableOpacity>
  );
}
