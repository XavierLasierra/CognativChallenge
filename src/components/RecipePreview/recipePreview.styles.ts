import { StyleSheet } from "react-native";

import Colors from "../../theme/Colors";
import Metrics from "../../theme/Metrics";

export default StyleSheet.create({
  imageContainer: {
    height: 128,
  },
  recipeImage: {
    flex: 1,
  },
  rowContainer: {
    alignItems: "stretch",
    backgroundColor: Colors.whiteFull,
    borderRadius: Metrics.borderRadius,
    justifyContent: "flex-start",
    marginBottom: Metrics.baseSpace,
    marginHorizontal: Metrics.baseSpace,
    overflow: "hidden",
  },
});
