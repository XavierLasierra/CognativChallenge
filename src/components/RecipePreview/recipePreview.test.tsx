import React from "react";
import RecipePreview from "./RecipePreview";
import { render, fireEvent, RenderAPI } from "@testing-library/react-native";
import recipesMock from "../../__mocks__/recipes.mock";
import { IRecipe } from "../../types/interfaces";

describe("Given a RecipePreview component", () => {
  describe("When it is rendered", () => {
    let screen: RenderAPI;
    let actionOnPress: () => void;
    let recipe: IRecipe;
    beforeEach(() => {
      actionOnPress = jest.fn();
      [recipe] = recipesMock;

      screen = render(
        <RecipePreview recipe={recipe} actionOnPress={actionOnPress} />,
      );
    });

    test("Then should match the snapshot", () => {
      expect(screen).toMatchSnapshot();
    });

    describe("And you click on it", () => {
      test("Then actionOnPress should have been called with the id of the recipe", () => {
        const id = recipe._id;
        const button = screen.getByRole("button");
        fireEvent(button, "press");

        expect(actionOnPress).toHaveBeenCalledWith(id);
      });
    });
  });
});
