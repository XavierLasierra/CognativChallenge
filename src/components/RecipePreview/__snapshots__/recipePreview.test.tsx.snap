// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RecipePreview component When it is rendered Then should match the snapshot 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "justifyContent": "flex-start",
        "marginBottom": 16,
        "marginHorizontal": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "height": 128,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://www.themealdb.com/images/media/meals/1520084413.jpg",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#9e9e9e",
            "fontSize": 10,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "letterSpacing": 1,
            "lineHeight": 24,
            "textAlign": "left",
          }
        }
      >
        Fish
      </Text>
      <Text
        style={
          Object {
            "color": "#333333",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "500",
            "letterSpacing": 0.5,
            "lineHeight": 24,
            "textAlign": "left",
          }
        }
      >
        Escovitch Fish
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            11
             minutes
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            Hard
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "marginRight": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#bdbdbd",
                "fontSize": 13,
                "fontStyle": "normal",
                "fontWeight": "300",
                "letterSpacing": 0.5,
                "lineHeight": 24,
              }
            }
          >
            3
             people
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
